### ISSUE TITLE
pushing pages which extend other pages gives error unless 'base' page is already on the server

### ISSUE THREAD

**Issue by exomarc** [locomotivecms/wagon#160]
I'm trying to push a page which extends another page.

index.liquid.haml
inner.liquid.haml --> extends index

I'm using the Git/master version of wagon and I'm pushing to LocomotiveHosting.

```
$ bundle exec wagon push hosting
[deprecated] I18n.enforce_available_locales will default to true in the future. If you really want to skip validation of your locale you can set I18n.enforce_available_locales = false to avoid this message.
* Pushing Site
* Pushing Snippets
  en
    updating Head...............................................................[done]
    updating Promo slider.......................................................[done]
* Pushing ContentTypes
    creating Home Slides........................................................[done]
    updating Messages...........................................................[done]
* Pushing Pages
  en
    updating index..............................................................[done]
    updating 404................................................................[done]
    creating contact-us      template => The page '' extends a template which does not exist

      raw_template => The page '' extends a template which does not exist
```

**Comment by abhishiv** [comment]
Just curious, was it fixed in a new version? I'm still getting this error on 1.5.2.

**Comment by did** [comment]
That should have been fixed in the 1.5.2 version. Sometimes, you have to set the position property of a page to make it pushed before another one at the same level.
Anyway, I'm working on the v3 version of the engine and I will get rid of the code which forces Wagon to push pages in a certain order. The order is given by both the tree structure of the site AND the template dependencies. Believe me, it's not trivial to solve the order to match these requirements. Hopefully, that will be removed in v3.
